x-pack/plugins/upgrade_assistant/public/application/app.tsx

```diff
@@ -52,94 +48,71 @@ const AppHandlingClusterUpgradeState: React.FunctionComponent = () => {
 
   if (missingManageSpacesPrivilege) {
     return (
-      <EuiPageContent
-        verticalPosition="center"
-        horizontalPosition="center"
-        color="subdued"
-        data-test-subj="missingKibanaPrivilegesMessage"
-      >
-        <NotAuthorizedSection
-          title={
-            <FormattedMessage
-              id="xpack.upgradeAssistant.app.deniedPrivilegeTitle"
-              defaultMessage="Kibana admin role required"
-            />
-          }
-          message={
-            <FormattedMessage
-              id="xpack.upgradeAssistant.app.deniedPrivilegeDescription"
-              defaultMessage="To use Upgrade Assistant and resolve deprecation issues, you must have access to manage all Kibana spaces."
-            />
-          }
-        />
-      </EuiPageContent>
+      <NotAuthorizedSection
```

Comment:
We'll need to support a `data-test-subj` here as well for testing purposes.

I would recommend adding support directly to the `es_ui_shared` component for this [here](https://github.com/elastic/kibana/blob/main/src/plugins/es_ui_shared/__packages_do_not_import__/authorization/components/not_authorized_section.tsx). The change would look something like this:

```
import React from 'react';
import { EuiEmptyPrompt } from '@elastic/eui';

interface Props {
  title: React.ReactNode;
  message: React.ReactNode | string;
  dataTestSubj?: string;
}

export const NotAuthorizedSection = ({ title, message, dataTestSubj }: Props) => (
  <EuiEmptyPrompt iconType="securityApp" data-test-subj={dataTestSubj ? dataTestSubj : 'notAuthorizedSection'}title={<h2>{title}</h2>} body={<p>{message}</p>} />
);

```

Then, you can pass `data-test-subj="missingKibanaPrivilegesMessage"` here in the Upgrade Assistant code.